### ISSUE TITLE
Help for WDRBD use

### ISSUE THREAD

**Issue by sungeun-kim** [mantechnology/wdrbd#1]
HI:

> ```
>   Thanks for your help.
>    I use drbdcon and this time, the time of sync process is much better!
> 
>    But question 2 is still when win8-2 reboot:
>    in win8-2
>           r0 role: Secondary
>               disk:Outdated
>               win8-1 connection: StandAlone
> 
>   Both nodes' firewall are totally OFF.
> 
>   Config file and log file are in attachment "WDRBD_HELP.rar"
> 
>   In win8-2's drbdlog, I see :
>  "The peer's disk size is too small! "
> 
>  win8-1's e driver is 21.40GB
>  win8-2's e driver is 21.41GB
>  Is this the reason? 
>  The SIZE of two disks use for mirror need to be TOTALLY same?
>  I remember drbd in linux,  different disk size between two mirror disks is acceptable. 
> 
>   Many Thanks!
> ```

At 2016-06-23 16:15:06, "Kim, Sung-eun" sekim@mantech.co.kr wrote:
hi, 

thanks for your concern.

plz, read inline comment for your questions.

설명: logo
Technical Research Center / DRBD Team
Principal Research Engineer
Sung-eun Kim               sekim@mantech.co.kr
202 Fl, Seoulforest Kolon digital Tower, 308-4, Seongsudong 2ga, Seongdong-gu, Seoul, Korea
Tel : 02-2136-6913 / Fax : 02-575-4858 / Call Center : 1833-7790
http://www.mantech.co.kr 
본문 이미지 1

2016-06-23 16:33 GMT+09:00 liangxin1300 liangxin1300@163.com:
hi dear mantech developers:

> ```
>     I'm a user from China.
>     It is a great honour to find WDRBD project in github ! Great job!
> 
>     I have met some problems when using WDBRD(  verson: drbd_windows-x64_1_1_2.exe; os: windows 2008 x64 on virtualbox). 
>    And I cant not understand Korean, so i cant not understand the doc in Korean.
>   ============================================================
>   Here are my questions:
>   QUESTION 1:
>          My mirror disk is driver E on both nodes, which size is 20G; 
>          I use command "drbdadm primary --force r0" to sync data firstly;
>          It's a long time to wait the number after "done" become "100.00". much longer than drbd in linux on same disk size;
>          So, what is the alternative way to sync data quickly? command "drbdadm new-current-uuid --clear-bitmap r0" when I can confirm data on both disks are the same?
> ```
> 
> first of all, you should disable TCP/IP nagle config. this is difference with linux drbd.
> 
> you can do that by "drbdcon /nagle_disable r0" (https://mantech.jira.com/wiki/display/WDRBDV9/drbdcon)

```
   =============================================================
```

> ```
>  QUESTION 2:
>    I have two nodes: win8-1 and win8-2;
>    Firstly sync between the tow nodes are successfully done, status are:
>    in win8-1
>         r0 role:Primary
>         disk:UpToDate
>         win8-2 role: Secondary
>             peer-disk: UpToDate
> 
>     Then I reboot win8-2, status are:
>     in win8-1
>           r0 role: Primary
>               disk:UpToDate
>               win8-2 connection: Connecting   
> 
>     in win8-2
>           r0 role: Secondary
>               disk:Outdated
>               win8-1 connection: StandAlone
> 
>    What should I do to make the disk state become UpToDate/UpToDate?
>    In linux, when I use drbd8.4.4 in this case, secondary node automatically sync from primary node and soon became UpToDate/UpToDate automatically.
> ```

```
   ============================================================
   Many thanks for your help!
```

in normal situation, after reboot, secondary will be connected automatically, and auto resync, finally become UpToDate/UpToDate. 
but your situation is abnormal. I think that after reboot, win8-2 try to connect to win8-1 , and at this point , some error is occurred, because win8-2 cstate is standalone.(maybe disconnected)

send us your config file, drbdlog(just type drbdlog in command line), and will help us.

thank you.

**Comment by sungeun-kim** [comment]
HI:
Thanks for your help.
I use drbdcon and this time, the time of sync process is much better!

But question 2 is still when win8-2 reboot:
in win8-2
r0 role: Secondary
disk:Outdated
win8-1 connection: StandAlone

Both nodes' firewall are totally OFF.

Config file and log file are in attachment "WDRBD_HELP.rar"

In win8-2's drbdlog, I see :
"The peer's disk size is too small! "

win8-1's e driver is 21.40GB
win8-2's e driver is 21.41GB
Is this the reason? 
The SIZE of two disks use for mirror need to be TOTALLY same?
I remember drbd in linux,  different disk size between two mirror disks is acceptable. 

Many Thanks!

**Comment by sungeun-kim** [comment]
Yes, you are right. currently, WDRBD have this issue. 
if two node's disk size is different, connection-fail occur.
we recommend that disk size is same.

another point is :
about your log(RegOpenKeyExW failed), you maybe run command line by user privilege. 
you should run command-line by admin privilege.

thank you

**Comment by sungeun-kim** [comment]
this issue will be resolved at WDRBD 1.2
thank you

**Comment by sungeun-kim** [comment]
in 1.1.8, this issue has resolved.
